### PR TITLE
MINIFICPP-850: Add operating system to c2 response

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
reviewers, please ignore white space via the review. My commit hook formatted this file after it detected some changes that would impact formatting.

**Comment by kevdoran** [comment]
Thanks, @phrocker. Will review

File: extensions/standard-processors/tests/unit/ManifestTests.cpp [hunk 1]
```diff
@@ -150,3 +151,24 @@ TEST_CASE("Test Scheduling Defaults", "[schedDef]") {
...
+  REQUIRE(serialized.size() > 0);
+  minifi::state::response::SerializedResponseNode proc_0;
+  for (const auto &node : serialized) {
+    if ("systemInfo" == node.name) {
+      for (const auto &sinfo : node.children) {
+        if ("operatingSystem" == sinfo.name) {
+          proc_0 = sinfo;
+        }
+      }
+
```
Comment (arpadboda) [inline extensions/standard-processors/tests/unit/ManifestTests.cpp]: break?

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 2]
```diff
@@ -279,46 +278,46 @@ class Device {
...
+        macs.insert(mac_add);
+        adapter = adapter->Next;
+      }
+    }
+
+    if (adapterPtr)
+    free(adapterPtr);
+    std::string macstr;
     for (auto &mac : macs) {
       macstr += mac;
```
Comment (arpadboda) [inline libminifi/include/core/state/nodes/DeviceInformation.h]: Do we just concat here without any separation?
Why doesn't this func simply return them in a vector?

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 3]
```diff
@@ -279,46 +278,46 @@ class Device {
...
     }
-	return macstr.length() > 0 ? std::to_string(hash_fn(macstr)) : "8675309";
-}
+    return macstr.length() > 0 ? std::to_string(hash_fn(macstr)) : "8675309";
```
Comment (arpadboda) [inline libminifi/include/core/state/nodes/DeviceInformation.h]: What's that magic number here? :)

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 4]
```diff
@@ -468,6 +470,26 @@ class DeviceInfoNode : public DeviceInformation {
...
 
+  /**
+   * Have found various ways of identifying different operating system variants
+   * so these were either pulled from header files or online.
+   */
+  static inline std::string getOperatingSystem() {
+#ifdef _WIN32
+    return "Windows 32";
+#elif _WIN64
+    return "Windows 64";
```
Comment (arpadboda) [inline libminifi/include/core/state/nodes/DeviceInformation.h]: This branch is dead in my opinion. I think _WIN32 is always defined when _WIN64 is, but I might be wrong.

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 5]
```diff
@@ -468,6 +470,26 @@ class DeviceInfoNode : public DeviceInformation {
...
+#elif __APPLE__ || __MACH__
```
Comment (arpadboda) [inline libminifi/include/core/state/nodes/DeviceInformation.h]: Afaik __MACH__ is not Apple-exclusive, but that most probably won't hurt.

File: libminifi/include/core/state/nodes/DeviceInformation.h (hunk 2 above)
Comment (phrocker) [reply libminifi/include/core/state/nodes/DeviceInformation.h]: We don't need to.

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 6]
```diff
@@ -196,11 +196,9 @@ class Device {
...
 
       if (family == AF_INET || family == AF_INET6) {
-        s = getnameinfo(ifa->ifa_addr,
-            (family == AF_INET) ? sizeof(struct sockaddr_in) :
-            sizeof(struct sockaddr_in6),
-            host, NI_MAXHOST,
-            NULL, 0, NI_NUMERICHOST);
+        s = getnameinfo(ifa->ifa_addr, (family == AF_INET) ? sizeof(struct sockaddr_in) : sizeof(struct sockaddr_in6), host, NI_MAXHOST,
+        NULL,
+                        0, NI_NUMERICHOST);
```
[the remark this answers is not in the document]
Comment (phrocker) [reply libminifi/include/core/state/nodes/DeviceInformation.h]: It's the formatter. Ignore whitesoaxe will show the no formatter changes

File: libminifi/include/core/state/nodes/DeviceInformation.h [hunk 7]
```diff
@@ -113,59 +113,59 @@ class Device {
   std::string device_id_;
  protected:
 
-	 std::vector<std::string> getIpAddresses() {
-		 static std::vector<std::string> ips;
-		 if (ips.empty()) {
+  std::vector<std::string> getIpAddresses() {
+    static std::vector<std::string> ips;
```
[the remark this answers is not in the document]
Comment (phrocker) [reply libminifi/include/core/state/nodes/DeviceInformation.h]: No it need not be

File: libminifi/include/core/state/nodes/DeviceInformation.h (hunk 4 above)
Comment (phrocker) [reply libminifi/include/core/state/nodes/DeviceInformation.h]: hmm, that's a good point. I'll make that a conditional Or. We can make them both windows. Arch, elsewhere sort of provides this already.

File: extensions/standard-processors/tests/unit/ManifestTests.cpp (hunk 1 above)
Comment (phrocker) [reply extensions/standard-processors/tests/unit/ManifestTests.cpp]: Oh yeah there's no reason to not stop. There won't be more than one but yah a break (doh). thanks!

**Pull request closed without merging**